packages/server/graphql/private/mutations/changeEmailDomain.ts

```diff
@@ -14,9 +15,33 @@ const changeEmailDomain: MutationResolvers['changeEmailDomain'] = async (
     throw new Error('New domain is the same as the old one')
   }
 
+  const [oldDomainUsers, newDomainUsers] = await Promise.all([
+    getUsersbyDomain(normalizedOldDomain),
+    getUsersbyDomain(normalizedNewDomain)
+  ])
+
+  if (!oldDomainUsers.length) {
+    throw new Error(`No users found with oldDomain: ${oldDomain}`)
+  }
+
+  const makeNewEmail = (email: string) => `${email.split('@')[0]}@${newDomain}`
+  const newDomainUserEmails = newDomainUsers.map((user) => user.email)
+  const proposedNewUserEmails = oldDomainUsers.map((user) => makeNewEmail(user.email))
+  const duplicateUserEmails = proposedNewUserEmails.filter((userEmail) =>
+    newDomainUserEmails.includes(userEmail)
+  )
+  const usersNotUpdatedIds = [] as string[]
```

Comment:
+1 L27-40 works, but it's a little hard to read.
Could you take a stab at simplifying it to make it a little more maintainable?

<details>
<summary> Click for Spoiler</summary>

```ts
const newDomainUserEmails = newDomainUsers.map((user) => user.email)
  const userIdsToUpdate = oldDomainUsers.filter(({email}) => !newDomainUserEmails.includes(email)).map(({id}) => id)
  const userIdsNotUpdated = oldDomainUsers.filter(({email}) => newDomainUserEmails.includes(email)).map(({id}) => id)
```
</details>

Reply:
Sure thing, updated! What do ya think?